HHS/Head-Start-TTADP
# [TTAHUB-1132] Add deleted (paranoid) statuses to goals and objectives

## Description of change

Retain deleted goals & objectives using Sequelize's paranoid deletion feature.

## How to test

The best way to confirm that records are being retained is to run the 'removes unused goals' test in the file goalServices/saveGoalForReport.js (removing the "force: true" from the cleanup .destroy in the afterAll) or follow the steps it details manually in the UI. The test asserts the goal disappears from the sequelize query results (it does, via the automatically included default scopes) but will be in the database with a value in the "deletedAt" column

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1132


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
